### ISSUE TITLE
Combination of brackets and functions inside ternary operator produce error

### ISSUE THREAD

**Issue by alexanderby** [microsoft/TypeScript#11582]
The following code produces error `Identifier expected`.

``` js
var x = true ? ({ y: function () { } }) : { y: function () { } };
```

Seems like TSLS thinks I'm declaring a type after `:`.

**Comment by Andarist** [comment]
@RyanCavanaugh this one works OK today, you can close the issue :)

**Comment by jakebailey** [comment]
Yes, this was fixed in #19979.
